presto-orc/src/test/java/com/facebook/presto/orc/metadata/statistics/AbstractStatisticsBuilderTest.java

```diff
@@ -65,47 +68,40 @@ public void testNoValue()
 
-        adder.accept(statisticsBuilder, expectedMax);
-        aggregateColumnStatistics.add(statisticsBuilder.buildColumnStatistics());
-        assertColumnStatistics(statisticsBuilder.buildColumnStatistics(), 3, expectedMin, expectedMax, aggregateColumnStatistics);
-
-        adder.accept(statisticsBuilder, expectedMax);
-        aggregateColumnStatistics.add(statisticsBuilder.buildColumnStatistics());
-        assertColumnStatistics(statisticsBuilder.buildColumnStatistics(), 4, expectedMin, expectedMax, aggregateColumnStatistics);
-
-        // add max then min
-        statisticsBuilder = statisticsBuilderSupplier.get();
-        aggregateColumnStatistics = new AggregateColumnStatistics();
+        // both
+        assertValues(expectedMin, expectedMax, ImmutableList.of(expectedMin, expectedMax));
+    }
 
-        adder.accept(statisticsBuilder, expectedMax);
-        aggregateColumnStatistics.add(statisticsBuilder.buildColumnStatistics());
-        assertColumnStatistics(statisticsBuilder.buildColumnStatistics(), 1, expectedMax, expectedMax, aggregateColumnStatistics);
+    public void assertValues(T expectedMin, T expectedMax, List<T> values)
+    {
+        assertValuesInternal(expectedMin, expectedMax, values);
+        assertValuesInternal(expectedMin, expectedMax, ImmutableList.copyOf(values).reverse());
 
-        adder.accept(statisticsBuilder, expectedMax);
-        aggregateColumnStatistics.add(statisticsBuilder.buildColumnStatistics());
-        assertColumnStatistics(statisticsBuilder.buildColumnStatistics(), 2, expectedMax, expectedMax, aggregateColumnStatistics);
+        List<T> randomOrder = new ArrayList<>(values);
+        Collections.shuffle(randomOrder, new Random(42));
+        assertValuesInternal(expectedMin, expectedMax, randomOrder);
+    }
 
-        adder.accept(statisticsBuilder, expectedMin);
+    private void assertValuesInternal(T expectedMin, T expectedMax, List<T> values)
+    {
+        B statisticsBuilder = statisticsBuilderSupplier.get();
+        AggregateColumnStatistics aggregateColumnStatistics = new AggregateColumnStatistics();
         aggregateColumnStatistics.add(statisticsBuilder.buildColumnStatistics());
-        assertColumnStatistics(statisticsBuilder.buildColumnStatistics(), 3, expectedMin, expectedMax, aggregateColumnStatistics);
+        assertColumnStatistics(statisticsBuilder.buildColumnStatistics(), 0, null, null, aggregateColumnStatistics);
 
-        adder.accept(statisticsBuilder, expectedMin);
-        aggregateColumnStatistics.add(statisticsBuilder.buildColumnStatistics());
-        assertColumnStatistics(statisticsBuilder.buildColumnStatistics(), 4, expectedMin, expectedMax, aggregateColumnStatistics);
+        for (int loop = 0; loop < 4; loop++) {
+            for (T value : values) {
+                adder.accept(statisticsBuilder, value);
+                aggregateColumnStatistics.add(statisticsBuilder.buildColumnStatistics());
```

Comment:
The tests you wrote have this interesting property that: for any two statistics object, one is a subset of the other. I don't think this is desirable.